File: source-codes/front-end/lewisglanzset-fe/src/app/service/contact.service.ts
import { Injectable } from '@angular/core';
import { ContactRest } from '../rest/contact.rest';
import { Contact } from '../model/contact.model';

@Injectable()
export class ContactService {

  constructor(private contactRest: ContactRest) {

  }

  public getAllContacts() {
    return this.contactRest.fetchAll();
  }

  public addContact(contact: Contact) {
    return this.contactRest.saveContact(contact);
  }

  public updateContact(contact: Contact) {
    return this.contactRest.updateContact(contact);
  }

  public deleteContact(contact: Contact) {
    return this.contactRest.deleteContact(contact);
  }

  public getContact(contactId: string) {
    return this.contactRest.fetchContact(contactId);
  }

}

File: source-codes/back-end/lewisglanzset/lgs-service/src/main/java/com/jpgl/lgs/service/ContactService.java
package com.jpgl.lgs.service;

import java.util.List;

import com.jpgl.lgs.common.model.Contact;

public interface ContactService {
    
    List<Contact> getAllContacts();
    
    boolean insertContact(Contact contact);
    
    boolean updateContact(Contact contact);
    
    boolean deleteContact(String contactId);

    Contact getContact(String contactId);
    
}
